mlir/test/Dialect/Tosa/constant-op-fold.mlir

```diff
@@ -114,6 +114,25 @@ func.func @transpose_nofold_quantized_types() -> tensor<1x1x2x2x!quant.uniform<i
 
 // -----
 
+// CHECK-LABEL: @transpose_dense_resource
+func.func @transpose_dense_resource() -> tensor<2x2xf32> {
+    %0 = "tosa.const"() <{value = dense_resource<resource> : tensor<2x2xf32>}> : () -> tensor<2x2xf32>
+    %1 = "tosa.const"() <{value = dense<[1, 0]> : tensor<2xi32>}> : () -> tensor<2xi32>
+
+    // CHECK: tosa.transpose
```

Comment:
I haven't work with `dense_resources` yet, but as you refactored the `transposeType` folder I would have assumed that this could be folded away if the resource is available. Otherwise, wouldn't it have sufficed to use `tryGetValues`?

Reply:
The refactoring was to clean up the logic so it had a single point of failure: the sequence of `tryGetValues` calls, I thought this made the code easier to reason about. As I mentioned in another comment, `tryGetValues` seems to fail for dense resources.